src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinValueInstantiator.kt

```diff
@@ -157,23 +120,19 @@ internal class KotlinValueInstantiator(
             numCallableParameters++
```

Comment:
Existing code, but I think this would be a bit clearer as `numCallableParameters + 1`, since we don't need to increment the value in place. That'll also allow `numCallableParameters` to be declared as a `val`

Reply:
I was planning to fix this when I introduce `ArgumentBucket` in the next PR.
https://github.com/FasterXML/jackson-module-kotlin/pull/439/files#diff-4fffadfbb02911c033bc87266a27253e0410d773ee8067b6f6786c440ac5306fR1

Reply:
Sounds good.